packages/instrumentation-pinecone/src/instrumentation.ts

```diff
@@ -0,0 +1,143 @@
+/*
+ * Copyright Traceloop
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+import type * as pinecone from "@pinecone-database/pinecone";
+
+import {
+  context,
+  trace,
+  Tracer,
+  SpanStatusCode,
+} from "@opentelemetry/api";
+import {
+  InstrumentationBase,
+  InstrumentationConfig,
+  InstrumentationModuleDefinition,
+  InstrumentationNodeModuleDefinition,
+  safeExecuteInTheMiddle,
+} from "@opentelemetry/instrumentation";
+import { SpanAttributes } from "@traceloop/ai-semantic-conventions";
+
+export class PineconeInstrumentation extends InstrumentationBase<any> {
+
+  constructor(config: InstrumentationConfig = {}) {
+    super("@traceloop/instrumentation-pinecone", "0.0.17", config);
+  }
+
+  public manuallyInstrument(
+    module: typeof pinecone & { openLLMetryPatched?: boolean },
+  ) {
+    if (module.openLLMetryPatched) {
+      return;
+    }
+    // TODO: Need to add code here?
+  }
```

Comment:
From what I can tell in the other instrumentations, the openAI instrumentation also calls the code in the patch method here but the llamaextract instrumentation just leaves this empty. What should I do for pinecone?